crypto/bn/bn_dh.c

```diff
@@ -194,6 +334,147 @@ static const BN_ULONG dh2048_256_q[] = {
     0xA709A097, 0x8CF83642
 };
 
+/* Primes from RFC 7919 */
+
+static const BN_ULONG ffdhe2048_p[] = {
+    0xFFFFFFFF, 0xFFFFFFFF, 0x61285C97, 0x886B4238, 0xC1B2EFFA, 0xC6F34A26,
```

Comment:
Why not U suffix here? Well, as just said type will be auto-selected to accommodate value, so that 0xFFFFFFFF is actually unsigned, but there might be an ambiguity of following character. BN is implemented under assumption that BN_ULONG can be 64 bits wide with THIRTY_TWO_BIT defined. There is no such platforms, but the assumption is still made. For reference, it's manifested by scrupulous masking with BN_MASK2. And ambiguity is in how compiler would treat 32-bit constant without U, when it's assigned to 64-bit value. Well, standard doesn't actually leaves ambiguity(*), but it's not insurance against compiler bugs. Or to summarize, wouldn't U suffix be appropriate here?

(*) 0xFFFFFFFF is supposed to be unsigned and no sign extension is supposed to take place)

Reply:
Well in both cases this follows the pattern that was already there for the (now inadvisable) RFC5114 parameters. Presumably if a compiler had thrown out a warning on those we'd know by now. But if you think it would be better if the RFC7919 (and earlier RFC5114) values should be changed I'll bow to your expertise.

Reply:
Note that bn_nist.c which has been about for a _very_ long time uses the same form. So we'd know about it by now if there were any problems.

Reply:
Note that I'm not saying that not having U suffix doesn't work, only wonder if it would make it less error-prone. It's also kind of a side note. Let us view it as possible overhaul for another time... As for it working elsewhere for long time. Well, somebody on platform where it *could* be a problem might never test it, because it falls out of application's scope... But one again, for some other time, maybe...